plugin/src/index.ts

```diff
@@ -115,14 +115,11 @@ The plugin no longer uses this and it should be deleted to avoid conflicts.\n`)
 
 export async function onPostBuild({
   constants: { PUBLISH_DIR, FUNCTIONS_DIST },
-  utils,
-}): Promise<void> {
+}: NetlifyPluginOptions): Promise<void> {
   if (shouldSkip(PUBLISH_DIR)) {
     return
   }
 
-  await saveCache({ publish: PUBLISH_DIR, utils })
```

Comment:
Now that we are moving the cache around, not just copying, if we only save the cache on successful builds are we essentially clearing the cache on failed build? Is it potentially worth keeping this save cache here?

Reply:
Thanks for catching this - I spent most time on this PR fixing tests (after not touching this repo for months) and this did slip. Overall I don't want to introduce changes to failed builds behavior, so I'm gonna investigate what is exact behavior now (wether we store or not cache on failed builds) and ensure this behavior is preserved

Reply:
Ok. `onPostBuild` is NOT executed on build failures - so moving this only to `onSuccess` (and not in `onError` or `onEnd`) is preserving current behavior.

There might be separate thing here to maybe start storing cache on failed builds, but this is out of scope for this PR.

Reply:
Ahh, I think I missed what the most important bit here - when we restore cache - we now move the files out and not just copy them - so if build fails - the cache location no longer has original cache files which would possibly result in subsequent build losing any of previous caches. The part I'm not sure about is wether we do upload that cache on failed builds and will look into that now

Reply:
It seems like we don't actually store cache from cache location on failed builds either, so this is still preserving the behavior.

BuildBot build command stage would fail on build failure and this would result in subsequent stages not running anymore (save cache is one of last stages after build command)